src/Oceananigans.jl

```diff
@@ -142,18 +141,31 @@ import Base:
     push!
 
 "Boolean denoting whether AMGX.jl can be loaded on machine."
-const hasamgx = @static Sys.islinux() ? true : false
+# TODO: find a way to check whether the libraries for AMGX and NETCDF 
+# (libamgxsh and libnetcdf, respectively) are installed on the machine
+const hasamgx   = false #@static Sys.islinux() ? true : false

```

Comment:
```suggestion
const hasamgx   = @static Sys.islinux() ? true : false

```

Reply:
We can't use this condition --- for example, MIT's Satori cluster is linux, but does not have AMGX (because it's a power 9 architecture). How do we correctly determine whether a user's architecture has AMGX?

Reply:
ok, I see...

@elise-palethorpe do you remember what else we had tried here?

Reply:
would a try catch work?

```julia
const hasamgx =
    try
        using AMGX
        true
    catch
        false
    end

Reply:
We tried that! It won't work.... because `using AMGX` errors **only** the first time it's called!

```Julia
(base) navid:OC.jl/ (glw/catke-plus-gm) $ julia --project                                                                                     [8:48:09]
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.2 (2022-09-29)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |

julia> using AMGX
ERROR: InitError: UndefVarError: libamgxsh not defined
Stacktrace:
 [1] getproperty
   @ ./Base.jl:31 [inlined]
 [2] __init__()
   @ AMGX ~/.julia/packages/AMGX/GFHHN/src/AMGX.jl:30
 [3] _include_from_serialized(pkg::Base.PkgId, path::String, depmods::Vector{Any})
   @ Base ./loading.jl:831
 [4] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String, build_id::UInt64)
   @ Base ./loading.jl:1039
 [5] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1315
 [6] _require_prelocked(uuidkey::Base.PkgId)
   @ Base ./loading.jl:1200
 [7] macro expansion
   @ ./loading.jl:1180 [inlined]
 [8] macro expansion
   @ ./lock.jl:223 [inlined]
 [9] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1144
during initialization of module AMGX

julia> using AMGX

julia> a = 1
1
```

Reply:
But what's the issue? Doesn't this still solve the problem?

Reply:
Are you saying that the problem is you can spuriously get `hasamgx=true` if you wrote `using AMGX` before `using Oceananigans`? I do agree that in this case, its wrong that `hasamgx=true`. On the other hand, it would seem that the situation is still happy, because one can use Oceananigans. If they _actually_ tried to use AMGX, they'd get another error (presumably). We could change the name of the variable to `definitely_doesnt_have_amgx` (with the inverse value) if want to be clear, like

```julia
const definitely_doesnt_have_amgx =
    try
        using AMGX
        false
    catch
        true
    end
```

Or am I missing something?